src/python/dxpy/dx_extract_utils/ExpressionInputsValidator.py

```diff
@@ -0,0 +1,164 @@
+from ..exceptions import err_exit
+
+
+class ExpressionInputsValidator:
+    """InputsValidator class for extract_assay expresion. Checks for invalid input combinations"""
+
+    def __init__(self, args, error_handler):
+        self.list_assays = args.list_assays
+        self.retrieve_expression_flag = args.retrieve_expression
+        self.assay_name = args.assay_name
+        self.input_json = args.input_json
+        self.input_json_file = args.input_json_file
+        self.json_help = args.json_help
+        self.sql = args.sql
+        self.additional_fields = args.additional_fields
+        self.additional_fields_help = args.additional_fields_help
+        self.expression_matrix = args.expression_matrix
+        self.delim = args.delim
+        self.output = args.output
+
+        self.error_handler = error_handler
+
+    def run_validations(self):
+        self.end_arguments_check("list_assays")
+        self.end_arguments_check("additional_fields_help")
+
+        if self.json_help and any(
+            [
+                self.assay_name,
+                self.sql,
+                self.additional_fields,
+                self.expression_matrix,
+                self.output,
+            ]
+        ):
+            self.error_handler = (
+                '"--json-help" cannot be passed with any of "--assay-name", "--sql", "--additional-fields", "--expression-matrix", or "--output"',
+            )
+
+        if self.retrieve_expression_flag and not any([self.input_json, self.json_help]):
+            self.error_handler = (
+                'The flag "--retrieve_expression" must be followed by a json input or json help.',
+            )
+
+        if self.input_json == "{}":
+            self.error_handler = (
+                'JSON for "--retrieve-expression" does not contain valid filter information.',
+            )
+
+    def end_arguments_check(self, end_arg):
+        args_specific_list = [
+            getattr(self, arg)
+            for arg in dir(self)
+            if not callable(getattr(self, arg))
+            and not arg.startswith("__")
+            and arg != end_arg
+        ]
+
+        invalid_combination = getattr(self, end_arg) and any(args_specific_list)
+
+        if invalid_combination:
+            self.error_handler = (
+                '"--{}" cannot be presented with other options.'.format(
+                    end_arg.replace("_", "-")
+                )
+            )
+
+    def error_exit(self):
```

Comment:
This method is not needed. `error_handler` is now a method and will be called directly within the class whenever needed.